applications/RANSApplication/python_scripts/formulations/turbulence_models/k_omega_rans_formulation.py

```diff
@@ -0,0 +1,142 @@
+    def GetSolvingVariable(self):
+        return KratosRANS.TURBULENT_KINETIC_ENERGY
+
+    def GetElementNamePrefix(self):
+        return "RansKOmegaK"
+
+    def GetConditionNamePrefix(self):
+        return ""
+
+
+class KOmegaOmegaRansFormulation(ScalarTurbulenceModelRansFormulation):
+    def GetSolvingVariable(self):
+        return KratosRANS.TURBULENT_SPECIFIC_ENERGY_DISSIPATION_RATE
+
+    def GetElementNamePrefix(self):
+        return "RansKOmegaOmega"
+
+    def GetConditionNamePrefix(self):
+        return "RansKOmegaOmega"
+
+
+class KOmegaRansFormulation(TwoEquationTurbulenceModelRansFormulation):
+    def __init__(self, model_part, settings):
+        default_settings = Kratos.Parameters(r'''
+        {
+            "formulation_name": "k_Omega",
+            "stabilization_method": "algebraic_flux_corrected",
+            "turbulent_kinetic_energy_solver_settings": {},
+            "turbulent_specific_energy_dissipation_rate_solver_settings": {},
+            "coupling_settings":
+            {
+                "relative_tolerance": 1e-3,
+                "absolute_tolerance": 1e-5,
+                "max_iterations": 10
+            },
+            "auxiliar_process_list": [],
+            "echo_level": 0
+        }''')
+
+        settings.ValidateAndAssignDefaults(default_settings)
+
+        super().__init__(
+            model_part,
+            settings,
+            KOmegaKRansFormulation(model_part, settings["turbulent_kinetic_energy_solver_settings"]),
+            KOmegaOmegaRansFormulation(model_part, settings["turbulent_specific_energy_dissipation_rate_solver_settings"]))
+
+    def AddVariables(self):
+        self.GetBaseModelPart().AddNodalSolutionStepVariable(Kratos.DENSITY)
+        self.GetBaseModelPart().AddNodalSolutionStepVariable(Kratos.VELOCITY)
+        self.GetBaseModelPart().AddNodalSolutionStepVariable(Kratos.MESH_VELOCITY)
+        self.GetBaseModelPart().AddNodalSolutionStepVariable(Kratos.NORMAL)
+        self.GetBaseModelPart().AddNodalSolutionStepVariable(Kratos.VISCOSITY)
+        self.GetBaseModelPart().AddNodalSolutionStepVariable(Kratos.KINEMATIC_VISCOSITY)
+        self.GetBaseModelPart().AddNodalSolutionStepVariable(Kratos.TURBULENT_VISCOSITY)
+        self.GetBaseModelPart().AddNodalSolutionStepVariable(KratosRANS.RANS_Y_PLUS)
+        self.GetBaseModelPart().AddNodalSolutionStepVariable(KratosRANS.TURBULENT_KINETIC_ENERGY)
+        self.GetBaseModelPart().AddNodalSolutionStepVariable(KratosRANS.TURBULENT_KINETIC_ENERGY_RATE)
+        self.GetBaseModelPart().AddNodalSolutionStepVariable(KratosRANS.TURBULENT_SPECIFIC_ENERGY_DISSIPATION_RATE)
+        self.GetBaseModelPart().AddNodalSolutionStepVariable(KratosRANS.TURBULENT_SPECIFIC_ENERGY_DISSIPATION_RATE_2)
+        self.GetBaseModelPart().AddNodalSolutionStepVariable(KratosRANS.RANS_AUXILIARY_VARIABLE_1)
+        self.GetBaseModelPart().AddNodalSolutionStepVariable(KratosRANS.RANS_AUXILIARY_VARIABLE_2)
```

Comment:
Do you require these to be historical?

Reply:
No I don't, These are only used in the adjoints, and in adjoints they need to be historical :/